codegen/src/main/twirl/templates/ScalaClient/Client.scala.txt

```diff
@@ -25,56 +26,30 @@ final class @{service.name}Client(settings: GrpcClientSettings)(implicit mat: Ma
   private val options = settings.options.getOrElse(CallOptions.DEFAULT)
 
   @for(method <- service.methods) {
-  def @{method.name}(in: @method.parameterType): @method.returnType = {
-  @if(method.methodType == akka.grpc.gen.Unary) {
-    ChannelApiHelpers.toScalaFuture(
-      ClientCalls.futureUnaryCall(channel.newCall(@{method.name}Descriptor, options), in)
-    )
-  }
-  @if(method.methodType == akka.grpc.gen.ServerStreaming){
-      val name = "@{service.grpcName}.@{method.grpcName}"
-      val flow =
-        ChannelApiHelpers.buildFlow[@method.inputTypeUnboxed, @method.outputTypeUnboxed](name) { responseObserver =>
-          new StreamObserver[@method.inputTypeUnboxed] {
-            override def onError(t: Throwable): Unit = responseObserver.onError(t)
-            override def onCompleted(): Unit = ()
-            override def onNext(request: @method.inputTypeUnboxed): Unit =
-              ClientCalls.asyncServerStreamingCall(
-                channel.newCall(@{method.name}Descriptor, options),
-                request,
-                responseObserver)
-          }
-        }
-
-
-
-    val bufferSize = options.getOption(CallOptions.Key.of("buffer_size", 10000))
-    Source
-      .single(in)
-      // channel calls don't support back-pressure so we need to buffered it
-      // and eventually fail the stream
-      .via(flow.buffer(bufferSize, OverflowStrategy.fail))
-  }
-  @if(method.methodType == akka.grpc.gen.ClientStreaming){
-    val name = "@{service.grpcName}.@{method.grpcName}"
-    val flow =
-      ChannelApiHelpers.buildFlow[@method.inputTypeUnboxed, @method.outputTypeUnboxed](name)  { responseObserver =>
-        ClientCalls.asyncClientStreamingCall(
-          channel.newCall(@{method.name}Descriptor, options),
-          responseObserver)
+  /**
+   * Lower level version of the method, prefer @{method.name}(@method.parameterType) if possible.
+   */
+  def @{method.name}(): RequestBuilder[@method.parameterType, @method.returnType] = {
+    val descriptor = @{method.name}Descriptor
+    @if(method.methodType == akka.grpc.gen.Unary) {
+      RequestBuilderImpl.unary(descriptor, channel, options)
```

Comment:
Do we want to organize the template so that the indentation of the generated code makes sense? No real preference here.

Reply:
Ah, I thought I did follow that, since it is how it was, but maybe having the templates readable is more important for maintainability than not having too many indentation levels in the generated sources?